### ISSUE TITLE
Prefer `path.resolve()` to `path.join(__dirname)`

### ISSUE THREAD

**Issue by fregante** [mysticatea/eslint-plugin-node#292]
It seems to me that `path.join(__dirname)` is equivalent to `path.resolve()`, so the latter would be preferred.

## Bad

```js
path.join(__dirname, 'dir/my-file.js')
```

## Good

```js
path.resolve('dir/my-file.js')
path.join('just', 'a/relative', '../path', 'my-file.js')
```

**Comment by cherryblossom000** [comment]
[`path.resolve`](https://nodejs.org/api/path.html#path_path_resolve_paths) resolves relative to the current working directory, which is not necessarily the same as `__dirname`:

> If, after processing all given `path` segments, an absolute path has not yet been generated, the current working directory is used.

**Comment by fregante** [comment]
Confirmed via 

```js
const path = require('path');
console.log(path.resolve(), 'resolve')
console.log(path.join(__dirname), 'join + dirname')
```
